sympy/matrices/expressions/tests/test_matpow.py

```diff
@@ -93,3 +93,29 @@ def test_doit_nested_MatrixExpr():
     Y = ImmutableMatrix([[2, 3], [4, 5]])
     assert MatPow(MatMul(X, Y), 2).doit() == (X*Y)**2
     assert MatPow(MatAdd(X, Y), 2).doit() == (X + Y)**2
+
+
+def test_identity_power():
+    k = Identity(n)
+    assert MatPow(k, 4).doit() == k
+    assert MatPow(k, 1).doit() == k
+    assert MatPow(k, -3).doit() == k
+    assert MatPow(k, 0).doit() == k
+    l = Identity(3)
+    assert MatPow(l, 2).doit() == l
+    assert MatPow(l, -1).doit() == l
+    assert MatPow(l, 0).doit() == l
+
+def test_zero_power():
+    z1 = ZeroMatrix(n, n)
+    assert MatPow(z1, 3).doit() == z1
+    raises(ValueError, lambda:MatPow(z1, -1).doit())
+    assert MatPow(z1, 0).doit() == Identity(n)
+    assert MatPow(z1, 2).doit() == z1
+    raises(ValueError, lambda:MatPow(z1, -2).doit())
+    z2 = ZeroMatrix(4, 4)
+    assert MatPow(z2, 4).doit() == z2
+    raises(ValueError, lambda:MatPow(z2, -3).doit())
+    assert MatPow(z2, 2).doit() == z2
+    assert MatPow(z2, 0).doit() == Identity(4)
+    raises(ValueError, lambda:MatPow(z2, -1).doit())
```

Comment:
I think "not", since i am seeing `O-|` this type of symbol on three 
lines at the end. Hovering over which gives "No New line at the end of 
file". I am not sure if that would result in an error on `python 
./bin/test sympy/utilities/tests/test_code_quality.py` (same before as 
i mentioned). Can you run `python ./bin/test 
sympy/utilities/tests/test_code_quality.py` on your machine locally?

On Sun, Aug 30, 2015 at 11:17 AM, Keval Shah <notifications@github.com> 
wrote:
> In sympy/matrices/expressions/tests/test_matpow.py:
> 
> > +    assert MatPow(l, -1).doit() == l
> > +    assert MatPow(l, 0).doit() == l
> > +
> > +def test_zero_power():
> > +    z1 = ZeroMatrix(n, n)
> > +    assert MatPow(z1, 3).doit() == z1
> > +    raises(ValueError, lambda:MatPow(z1, -1).doit())
> > +    assert MatPow(z1, 0).doit() == Identity(n)
> > +    assert MatPow(z1, 2).doit() == z1
> > +    raises(ValueError, lambda:MatPow(z1, -2).doit())
> > +    z2 = ZeroMatrix(4, 4)
> > +    assert MatPow(z2, 4).doit() == z2
> > +    raises(ValueError, lambda:MatPow(z2, -3).doit())
> > +    assert MatPow(z2, 2).doit() == z2
> > +    assert MatPow(z2, 0).doit() == Identity(4)
> > +    raises(ValueError, lambda:MatPow(z2, -1).doit())
> Ohh I get it now.
> Does everything look good after the last commit?
> 
> —
> Reply to this email directly or view it on GitHub.
>